src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -128,31 +128,26 @@ public boolean unlock(UUID snapshotId, String flightId) {
     public UUID createAndLock(Snapshot snapshot, String flightId) {
         logger.debug("createAndLock snapshot " + snapshot.getName());
 
-        String sql = "INSERT INTO snapshot (name, description, profile_id, flightid) " +
-            "VALUES (:name, :description, :profile_id, :flightid) ";
+        String sql = "INSERT INTO snapshot (name, description, profile_id, id, flightid) " +
+            "VALUES (:name, :description, :profile_id, :id, :flightid) ";
         MapSqlParameterSource params = new MapSqlParameterSource()
             .addValue("name", snapshot.getName())
             .addValue("description", snapshot.getDescription())
             .addValue("profile_id", snapshot.getProfileId())
+            .addValue("id", snapshot.getId())
             .addValue("flightid", flightId);
-        DaoKeyHolder keyHolder = new DaoKeyHolder();
         try {
-            jdbcTemplate.update(sql, params, keyHolder);
+            jdbcTemplate.update(sql, params);
```

Comment:
The duplicate key exception below might happen either because of the name or because of the id now. So may make the message a little more generic and include both the name and the id?